### PR TITLE
feat: collect jira development panel

### LINKED ISSUE #4304: [Feature][Jira] Collect data in jira development panel
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

In our jira use case, we are not link commit to issue by remotelinks, but link it by development panel with a jira plugin.  
We want data in development panel, can you collect it?  
The most useful data is the related commits. And it's better if collecting branches, pull requests and builds.

### data in development panel
image  
image

### Description

Now we can get commits related to issue in `issue_commits` entity.  
In jira data source, `issue_commits` is resolved by `_tool_jira_remotelinks` with a configuration of `remotelinkCommitShaPattern`.  
We have another way to link issue commits, can you support it?

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mindlesscloud**
### Summary
Fulfill #4304, provide Jira development panel collector and extractor, which will be activated only if the `JiraOptions.ApplicationType` is not empty.
- add `repo_url` to `_tool_jira_issue_commits`
- introduce a new field `ApplicationType` to `JiraOptions`, which will be taken as a request param during collecting development panel
- collect the development panel from the API `https://jirahost/rest/dev-status/1.0/issue/detail?issueId=25184&applicationType=GitLab&dataType=repository` and save to `_raw_jira_api_development_panels`
- extract `commit_url`, `commit_sha` and `repo_url` from raw layer and save to `_tool_jira_issue_commits`
- provide an API to query application types
- remove two redundant CSV files

### Does this close any open issues?
Closes #4304 

### Screenshots
image
image
image

### CODE REVIEW

**Review by narrowizard — CHANGES_REQUESTED** [review]

File: backend/plugins/jira/tasks/task_data.go [hunk 1]
```diff
@@ -110,6 +110,8 @@ type JiraOptions struct {
 	ScopeId              string
 	TransformationRuleId uint64
 	PageSize             int
+	// ApplicationType only used by development panel collector
+	ApplicationType string
```
[the remark this answers is not in the document]
Comment (mindlesscloud) [reply backend/plugins/jira/tasks/task_data.go:114]: Yes, for most scenarios, one is enough. Besides, the Jira development panel API is very time costing, we should minimize the number of HTTP requests as much as possible.

File: backend/plugins/jira/tasks/issue_repo_commit_convertor.go [hunk 2]
```diff
@@ -97,6 +97,7 @@ func ConvertIssueRepoCommits(taskCtx plugin.SubTaskContext) errors.Error {
 			item := &crossdomain.IssueRepoCommit{
 				IssueId:   issueIdGenerator.Generate(connectionId, issueCommit.IssueId),
 				CommitSha: issueCommit.CommitSha,
+				RepoUrl:   issueCommit.RepoUrl,
```
Comment (abeizn) [inline backend/plugins/jira/tasks/issue_repo_commit_convertor.go:100]: Some parameters are missing assignments：host, namespace, repo_name
image

Comment (mindlesscloud) [reply backend/plugins/jira/tasks/issue_repo_commit_convertor.go:100]: they are filled by invoking the function `api.RefineIssueRepoCommit`

**Review by abeizn — APPROVED** [review]
LGTM

**Pull request merged**